File: app/src/main/java/bassamalim/hidaya/receivers/DeviceBootReceiver.java
package bassamalim.hidaya.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import bassamalim.hidaya.other.Global;

public class DeviceBootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        Log.i(Global.TAG, "in device boot receiver");
        if (intent.getAction().equals("android.intent.action.BOOT_COMPLETED")) {
            Intent intent1 = new Intent(context, DailyUpdateReceiver.class);
            intent1.setAction("boot");
            context.sendBroadcast(intent1);
        }
    }
}

File: app/src/main/java/bassamalim/hidaya/receivers/DeviceBootReceiver.kt
package bassamalim.hidaya.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import bassamalim.hidaya.other.Global

class DeviceBootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        Log.i(Global.TAG, "in device boot receiver")

        if (intent.action == "android.intent.action.BOOT_COMPLETED") {
            val intent1 = Intent(context, DailyUpdateReceiver::class.java)
            intent1.action = "boot"
            context.sendBroadcast(intent1)
        }
    }

}